lib/node_modules/@stdlib/repl/lib/syntax_highlighter.js

```diff
@@ -0,0 +1,154 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable no-underscore-dangle, no-restricted-syntax, no-invalid-this */
+
+'use strict';
+
+// MODULES //
+
+var readline = require( 'readline' );
+var logger = require( 'debug' );
+var setNonEnumerableReadOnly = require( '@stdlib/utils/define-nonenumerable-read-only-property' );
+var tokenizer = require( './tokenizer.js' );
+var THEMES = require( './themes.js' );
+var ANSI = require( './ansi_colors.js' );
+
+
+// VARIABLES //
+
+var debug = logger( 'repl:syntax-highlighter' );
+
+
+// MAIN //
+
+/**
+* Constructor for creating a syntax-highlighter.
+*
+* @private
+* @constructor
+* @param {repl} repl - REPL instance
+* @param {WritableStream} ostream - writable stream
+* @returns {SyntaxHighlighter} syntax-highlighter instance
+*/
+function SyntaxHighlighter( repl, ostream ) {
+	if ( !( this instanceof SyntaxHighlighter ) ) {
+		return new SyntaxHighlighter( repl, ostream );
+	}
+	debug( 'Creating a new syntax-highlighter' );
+
+	// Cache a reference to the provided REPL instance:
+	this._repl = repl;
+
+	// Cache a reference to readline interface:
+	this._rli = repl._rli;
+
+	// Cache a reference to the output writable stream:
+	this._ostream = ostream;
+
+	// Cache a reference to the available themes:
+	this._themes = THEMES;
+
+	// Initialize a buffer containing the current theme:
+	this._theme = 'default';
```

Comment:
Calling this a "buffer" is misleading. And also, this will need to be rethought once a setting is added to the REPL, as the current setting value will need to be queried before highlighting after each keypress.

Reply:
Yes, I had that in mind, we will probably directly query the setting (when it's added) instead of storing it with the syntax highlighter instance